File: client/src/app/_models/sportparams.ts
export class SportParams {
    pageNumber = 1;
    pageSize = 10;
    sportType: string | null = null;
    dateFrom: string | null = null;
    dateTo: string | null = null;
    distanceFrom: number | null = null;
    distanceTo: number | null = null;
    orderBy = 'date';
}

File: SportAndStepsApps/Helpers/SportParams.cs
namespace SportAndStepsApps.Helpers;

public class SportParams
{
    private const int MaxPageSize = 20;
    public int PageNumber { get; set; } = 1;
    private int pageSize = 10;
    public int PageSize
    {
        get => pageSize;
        set => pageSize = (value > MaxPageSize) ? MaxPageSize : value;
    }
    public string? SportType { get; set; }
    public int? DistanceFrom { get; set; }
    public int? DistanceTo { get; set; }
    public DateTime? DateFrom { get; set; }
    public DateTime? DateTo { get; set; }

    public string OrderBy { get; set; } = "date";
}
